torch/distributed/fsdp/_fully_shard/_fsdp_collectives.py

```diff
@@ -437,6 +438,18 @@ def foreach_reduce(
                 )
                 all_reduce_input = reduce_output
                 all_reduce_event = all_reduce_stream.record_event()
+    # -- END: ops in reduce_scatter stream
+
+    if all_reduce_hook is not None:
+        # Execute user-specified all reduce hook across FSDP units.
```

Comment:
super nit: the `across FSDP units` part is a bit confusing to me since the 'across' could mean both "same FSDP unit but across DP ranks" or "same DP rank but across FSDP units on this rank"

maybe can just exclude the "across FSDP units" part -- it should be clear the all-reduce will be across some DP ranks

Reply:
Understood. Naming is hard :(
Updated.